mlflow/spark.py

```diff
@@ -759,15 +809,26 @@ def predict(self, pandas_df):
         """
         from pyspark.ml import PipelineModel
 
-        spark_df = self.spark.createDataFrame(pandas_df)
+        spark_df = find_and_set_features_col_as_vector_if_needed(
+            self.spark.createDataFrame(pandas_df), self.spark_model
+        )
+        prediction_column = "prediction"
         if isinstance(self.spark_model, PipelineModel) and self.spark_model.stages[-1].hasParam(
             "outputCol"
         ):
-            # make sure predict work by default for Transformers
-            self.spark_model.stages[-1].setOutputCol("prediction")
+            from pyspark.sql import SparkSession
+
+            spark = SparkSession.builder.getOrCreate()
+            # do a transform with an empty input DataFrame
+            # to get the schema of the transformed DataFrame
+            transformed_df = self.spark_model.transform(spark.createDataFrame([], spark_df.schema))
+            # Ensure prediction column doesn't already exist
+            if prediction_column not in transformed_df.columns:
```

Comment:
What was the previous behavior when the `"prediction"` column already existed? Did we throw? We should make sure to preserve the previous behavior to avoid breaking changes.

Reply:
yes, it threw `java.lang.IllegalArgumentException` exception, but it can only occur if the transformer is a pipeline type and if the last stage of that pipeline is not a Model transformer such as an `IndexToString` stage, I added unit tests fot it [here](https://github.com/mlflow/mlflow/pull/5719/files#diff-317be50f891ddc0e3a113301f7db84008e7d35818b097bfc4668958465661005R1009) and [here](https://github.com/mlflow/mlflow/pull/5719/files#diff-317be50f891ddc0e3a113301f7db84008e7d35818b097bfc4668958465661005R1023).
Considering that and knowing it threw an exception, this method was basically of no use currently if someone had a pipeline with an `IndexToString` stage in it. Do you still think we should continue to throw an exception and add this change in the next major version ?

Reply:
Ah, got it. If the previous throwing behavior was purely a limitation and we're removing the limitation here by expanding the supported set of pipelines, I think this is great. Thanks Jas!